File: Count Elements With Maximum Frequency.py
class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        max_frequency = 0
        sum_frequency = 0
        freq_dict = {}

        for num in nums:
            freq_dict[num] = freq_dict.get(num, 0) + 1
            current_frequency = freq_dict[num]

            if current_frequency > max_frequency:
                max_frequency = current_frequency
                sum_frequency = current_frequency
            elif current_frequency == max_frequency:
                sum_frequency += current_frequency

        return sum_frequency  

File: count-elements-with-maximum-frequency.java
import java.util.*;

class Solution {
    public int maxFrequencyElements(int[] nums) {
        int[] freq = new int[101];  // numbers are in range 1–100
        for (int num : nums) {
            freq[num]++;
        }

        int maxFreq = 0;
        for (int f : freq) {
            maxFreq = Math.max(maxFreq, f);
        }

        int result = 0;
        for (int f : freq) {
            if (f == maxFreq) {
                result += f;
            }
        }

        return result;
    }
}
